### ISSUE TITLE
KeyError: "Unable to open object (object '4.000' doesn't exist)"

### ISSUE THREAD

**Issue by jhkang727** [Yuxuan-W/GEB-Plus#1]
Hello, while running this code I had an error.
The error message is as following 

Traceback (most recent call last):
  File "/home/S20225009/Challenge_track2/GEBC-master/run_captioning.py", line 479, in <module>
    main()
  File "/home/S20225009/Challenge_track2/GEBC-master/run_captioning.py", line 455, in main
    last_checkpoint = train(args, train_dataloader, val_dataloader, model, tokenizer)
  File "/home/S20225009/Challenge_track2/GEBC-master/run_captioning.py", line 172, in train
    for step, (boundary_ids, batch) in enumerate(train_dataloader):
  File "/home/S20225009/.conda/envs/LOVEU2/lib/python3.9/site-packages/torch/utils/data/dataloader.py", line 435, in __next__
    data = self._next_data()
  File "/home/S20225009/.conda/envs/LOVEU2/lib/python3.9/site-packages/torch/utils/data/dataloader.py", line 1085, in _next_data
    return self._process_data(data)
  File "/home/S20225009/.conda/envs/LOVEU2/lib/python3.9/site-packages/torch/utils/data/dataloader.py", line 1111, in _process_data
    data.reraise()
  File "/home/S20225009/.conda/envs/LOVEU2/lib/python3.9/site-packages/torch/_utils.py", line 428, in reraise
    raise self.exc_type(msg)
KeyError: Caught KeyError in DataLoader worker process 15.
Original Traceback (most recent call last):
  File "/home/S20225009/.conda/envs/LOVEU2/lib/python3.9/site-packages/torch/utils/data/_utils/worker.py", line 198, in _worker_loop
    data = fetcher.fetch(index)
  File "/home/S20225009/.conda/envs/LOVEU2/lib/python3.9/site-packages/torch/utils/data/_utils/fetch.py", line 44, in fetch
    data = [self.dataset[idx] for idx in possibly_batched_index]
  File "/home/S20225009/.conda/envs/LOVEU2/lib/python3.9/site-packages/torch/utils/data/_utils/fetch.py", line 44, in <listcomp>
    data = [self.dataset[idx] for idx in possibly_batched_index]
  File "/home/S20225009/Challenge_track2/GEBC-master/datasets/captioning_dataset.py", line 81, in __getitem__
    frame_scale, frame_feat, obj_bbox, obj_feat = self.get_regions(video_id, timestamps)
  File "/home/S20225009/Challenge_track2/GEBC-master/datasets/captioning_dataset.py", line 101, in get_regions
    boundary=[h5[f"{sampled_time['boundary']:.3f}"]['feature'][0]],
  File "h5py/_objects.pyx", line 54, in h5py._objects.with_phil.wrapper
  File "h5py/_objects.pyx", line 55, in h5py._objects.with_phil.wrapper
  File "/home/S20225009/.conda/envs/LOVEU2/lib/python3.9/site-packages/h5py/_hl/group.py", line 305, in __getitem__
    oid = h5o.open(self.id, self._e(name), lapl=self._lapl)
  File "h5py/_objects.pyx", line 54, in h5py._objects.with_phil.wrapper
  File "h5py/_objects.pyx", line 55, in h5py._objects.with_phil.wrapper
  File "h5py/h5o.pyx", line 190, in h5py.h5o.open
KeyError: "Unable to open object (object '4.000' doesn't exist)"

do you know what this error means?
Ps. the dataset was fully downloaded properly

**Comment by Yuxuan-W** [comment]
Hi, could you please provide the corresponding video id where the error happens? I will check the h5 file.

**Comment by jhkang727** [comment]
Hello, sorry for the late reply.

We have removed the problem IDs from trainset_highest_f1.json file and running the code, and the IDs identified so far are as follows.

, "ya63SMpFrJ0": [{"boundary_id": "ya63SMpFrJ0_0", "timestamp": 3.74348, "prev_timestamp": 0, "next_timestamp": 5.35225, "label": "Change of Action", "subject": "man in white shirt blue and black stripes tie", "status_before": "tying the tie with hand", "status_after": "pull the tie towards downward", "action_of_cause": null, "caption": "Subject: man in white shirt blue and black stripes tie //Status_Before: tying the tie with hand //Status_After: pull the tie towards downward"}, {"boundary_id": "ya63SMpFrJ0_1", "timestamp": 5.35225, "prev_timestamp": 3.74348, "next_timestamp": 7.58346, "label": "Change of Action", "subject": "man in white shirt blue and black stripes tie", "status_before": "tying the tie with hand", "status_after": "pull the tie towards downward", "action_of_cause": null, "caption": "Subject: man in white shirt blue and black stripes tie //Status_Before: tying the tie with hand //Status_After: pull the tie towards downward"}, {"boundary_id": "ya63SMpFrJ0_2", "timestamp": 7.58346, "prev_timestamp": 5.35225, "next_timestamp": 8.83792, "label": "Change of Action", "subject": "man in white shirt blue and black stripes tie", "status_before": "tying the tie with hand", "status_after": "pull the tie towards downward", "action_of_cause": null, "caption": "Subject: man in white shirt blue and black stripes tie //Status_Before: tying the tie with hand //Status_After: pull the tie towards downward"}, {"boundary_id": "ya63SMpFrJ0_3", "timestamp": 8.83792, "prev_timestamp": 7.58346, "next_timestamp": 9.978, "label": "Change of Action", "subject": "man in white shirt blue and black stripes tie", "status_before": "tying the tie with hand", "status_after": "pull the tie towards downward", "action_of_cause": null, "caption": "Subject: man in white shirt blue and black stripes tie //Status_Before: tying the tie with hand //Status_After: pull the tie towards downward"}]

, "yIlUYJPPGOc": [{"boundary_id": "yIlUYJPPGOc_0", "timestamp": 6.70464, "prev_timestamp": 0, "next_timestamp": 7.86848, "label": "Change of Subject", "subject": "person in blue dress", "status_before": "/0", "status_after": "/1", "action_of_cause": "came into the scene", "caption": "Subject: person in blue dress //Status_Before: /0 //Status_After: /1"}, {"boundary_id": "yIlUYJPPGOc_1", "timestamp": 7.86848, "prev_timestamp": 6.70464, "next_timestamp": 9.977, "label": "Change of Subject", "subject": "person in blue dress", "status_before": "/1", "status_after": "/0", "action_of_cause": "move out of the scene", "caption": "Subject: person in blue dress //Status_Before: /1 //Status_After: /0"}]

"yIo2f9kqJb0": [{"boundary_id": "yIo2f9kqJb0_0", "timestamp": 2.72129, "prev_timestamp": 0, "next_timestamp": 4.35751, "label": "Change of Action", "subject": "human hand with gold ring", "status_before": "holding the book with both the hands", "status_after": "turn the page of the book with the hand on the left", "action_of_cause": null, "caption": "Subject: human hand with gold ring //Status_Before: holding the book with both the hands //Status_After: turn the page of the book with the hand on the left"}, {"boundary_id": "yIo2f9kqJb0_1", "timestamp": 4.35751, "prev_timestamp": 2.72129, "next_timestamp": 5.78705, "label": "Change of Action", "subject": "kid in red shirt", "status_before": "sitting and holding the page of the book with both the hands", "status_after": "turn the page of the book with the hand on the right", "action_of_cause": null, "caption": "Subject: kid in red shirt //Status_Before: sitting and holding the page of the book with both the hands //Status_After: turn the page of the book with the hand on the right"}, {"boundary_id": "yIo2f9kqJb0_2", "timestamp": 5.78705, "prev_timestamp": 4.35751, "next_timestamp": 7.52661, "label": "Change of Action", "subject": "human hand with gold ring", "status_before": "sitting by keeping hand on the left on the book", "status_after": "take off the hand on the left from the book", "action_of_cause": null, "caption": "Subject: human hand with gold ring //Status_Before: sitting by keeping hand on the left on the book //Status_After: take off the hand on the left from the book"}, {"boundary_id": "yIo2f9kqJb0_3", "timestamp": 7.52661, "prev_timestamp": 5.78705, "next_timestamp": 8.24999, "label": "Change of Action", "subject": "human hand with gold ring", "status_before": "taking off the hand on the left from the book", "status_after": "shift the hand on the left downward on the book", "action_of_cause": null, "caption": "Subject: human hand with gold ring //Status_Before: taking off the hand on the left from the book //Status_After: shift the hand on the left downward on the book"}, {"boundary_id": "yIo2f9kqJb0_4", "timestamp": 8.24999, "prev_timestamp": 7.52661, "next_timestamp": 9.967, "label": "Change of Action", "subject": "human hand with gold ring", "status_before": "shifting the hand on the left downward on the book", "status_after": "take off the hand on the left from the book", "action_of_cause": null, "caption": "Subject: human hand with gold ring //Status_Before: shifting the hand on the left downward on the book //Status_After: take off the hand on the left from the book"}], "z2ySm6Vbuf8": [{"boundary_id": "z2ySm6Vbuf8_0", "timestamp": 6.5156, "prev_timestamp": 0, "next_timestamp": 9.967, "label": "Change of Action", "subject": "human in blue jeans", "status_before": "sitting on the chair and cleaning the part with air pipe", "status_after": "raise the part", "action_of_cause": null, "caption": "Subject: human in blue jeans //Status_Before: sitting on the chair and cleaning the part with air pipe //Status_After: raise the part"}],

We don’t know if it is all or there’s more of them.

Sincerely,
Jeonghun Kang.

**Comment by Yuxuan-W** [comment]
Hi I checked my dropbox folder and found these vids missed, there must be some error when uploading. Sorry for that and I upload the feature again to google drive, here's the link:

https://drive.google.com/drive/folders/1v77sWxRa_TpIQdlcXKyrGvH_bLU7ZP7f?usp=sharing

I'm very sorry for this mistake. In this google drive folder I found all the missed vids feature, I will also update the link on our website. (Never gonna trust dropbox again...
